packages/calypso-e2e/src/lib/components/sidebar-component.ts

```diff
@@ -63,6 +63,15 @@ export class SidebarComponent {
 			await Promise.all( [ this.page.waitForNavigation(), this.page.click( itemSelector ) ] );
 		}
 
+		/**
+		 * Do not attempt to retry if we've navigated outside wp.com as the sidebar
+		 * is no longer present.
+		 */
+		const currentURL = await this.page.url();
+		if ( ! currentURL.startsWith( 'https://wordpress.com' ) ) {
```

Comment:
I think this is a good trade-off, but we should be aware that it'll eventually prevent us from running CI tests against a different URL, not a huge deal now as we haven't really done that yet, but there is a parameter for that in the `WPCom Tests->Gutenberg tests` `buildType`, called `Test URL`. cc @scinos